traffic_ops/app/lib/API/DeliveryServiceServer.pm

```diff
@@ -80,15 +86,21 @@ sub assign_servers_to_ds {
 		return $self->forbidden();
 	}
 
-	if ( ref($servers) ne 'ARRAY' ) {
-		return $self->alert("Servers must be an array");
-	}
-
 	my $ds = $self->db->resultset('Deliveryservice')->find( { id => $ds_id } );
 	if ( !defined($ds) ) {
 		return $self->not_found();
 	}
 
+	my $tenant_utils = Utils::Tenant->new($self);
+	my $tenants_data = $tenant_utils->create_tenants_data_from_db();
+	if (!$tenant_utils->is_ds_resource_accessible($tenants_data, $ds->tenant_id)) {
+		return $self->alert("Invalid delivery-service assignment. The delivery-service is not avaialble for your tenant.");
```

Comment:
Are you sure?
This is not part of the path

$r->post("/api/$version/deliveryserviceserver")->over( authenticated => 1, not_ldap => 1 )->to( 'DeliveryServiceServer#assign_servers_to_ds', namespace => $namespace );